api/src/main/java/org/pmiops/workbench/workspaces/WorkspacesController.java

```diff
@@ -204,7 +204,7 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
   }
 
   @Override
-  public ResponseEntity<Workspace> createWorkspace(Workspace workspace) {
+  public ResponseEntity<Workspace> createWorkspace(Workspace workspace) throws BadRequestException {
```

Comment:
Not sure how much value there is in annotating this on controller level methods. My assumption is that every controller method can throw all of the HTTP exceptions.

We rarely invoke our controller methods directly so this feels more like a comment or documentation. We already have a place for documenting controller method exceptions in our swagger.

Reply:
I agree in general, but WorkspacesController isn't a standard controller. It's used internally as a service, and has enough inconsistent behavior that it's worth calling out. I generally don't go to the yaml file to look for (unchecked) exceptions.

The root of all this is that exceptions making up part of method contracts should ideally be checked. (I believe that's not possible with the swagger-generated api definitions).

I started putting these on after tripping over an unexpected exception for the umpteenth time. It's definitely hacky and ugly, but it can make debugging things a bit easier to pretend some of these are checked.